libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -1323,14 +1322,32 @@ void AC_PosControl::get_stopping_point_U_cm(postype_t &stopping_point_u_cm) cons
 /// get_bearing_to_target_cd - get bearing to target position in centi-degrees
 int32_t AC_PosControl::get_bearing_to_target_cd() const
 {
-    return get_bearing_cd(_inav.get_position_xy_cm(), _pos_target_neu_cm.tofloat().xy());
+    return get_bearing_cd(Vector2f{0.0, 0.0}, (_pos_target_neu_cm.xy() - _pos_estimate_ned_m.xy() * 100.0).tofloat());
 }
 
 
 ///
 /// System methods
 ///
 
+
+void AC_PosControl::update_estimates(bool high_vibes)
+{
+    if (!AP::ahrs().get_relative_position_NED_origin(_pos_estimate_ned_m)) {
+        float posD;
+        if (AP::ahrs().get_relative_position_D_origin_float(posD)) {
```

Comment:
Isn't this the same as the pos D that returned in the vector from the previous call?

Reply:
No it isn't, it has an internal redundancy.
This is something that we may be able to improve.

Reply:
I have tried to explain this in the comment:

// Updates internal position and velocity estimates in the NED frame.
// Falls back to vertical-only estimates if full NED data is unavailable.
// When high_vibes is true, forces use of vertical fallback for velocity.